### PR TITLE
Wonder Guard should take effect on turn 2 of Sky Drop

### PR DESCRIPTION
**Pull request by Volco**
https://github.com/Zarel/Pokemon-Showdown/projects/3#card-25870318

### CODE REVIEW

**Comment by Zarel** [comment]
I wonder if this should work like Kee Berry, where we should consider Sky Drop a status move on its first turn?

Incidentally: Is Sky Drop Kee Berry bugged?

**Comment by Volco** [comment]
Good question, i could test that

**Comment by Volco** [comment]
after testing it does not seem to trigger 
image

**Comment by scheibo** [comment]
Would you please consider adding new unit tests for this change and the others you are proposing? Unit testing mechanics issues should be quite straightforward and helps prevent regressions going forward.

**Comment by Volco** [comment]
okay, i'll take a look at how those work and then update the prs

**Comment by Volco** [comment]
I'm not quite sure if i did the test correctly so if someone could take a look at if and tell me if anything needs to be changed i'd appreciate it

File: test/sim/abilities/wonderguard.js [hunk 1]
```diff
@@ -43,4 +43,12 @@ describe('Wonder Guard', function () {
 		battle.setPlayer('p2', {team: [{species: "Reshiram", ability: 'turboblaze', moves: ['fusionflare']}]});
 		assert.hurts(battle.p1.active[0], () => battle.makeChoices('move sleeptalk', 'move fusionflare'));
 	});
+
+	it('should allow first turn of Sky Drop but not the second', function () {
+		battle = common.createBattle();
+		battle.setPlayer('p1', {team: [{species: "Smeargle", ability: 'wonderguard', moves: ['sleeptalk']}]});
+		battle.setPlayer('p2', {team: [{species: "Aerodactyl", ability: 'owntempo', moves: ['skydrop']}]});
+		battle.makeChoices('move sleeptalk', 'move skydrop');
+		assert.false.hurts(battle.p1.active[0], () => battle.makeChoices('move sleeptalk', 'move skydrop'));
```
Comment (Zarel) [inline test/sim/abilities/wonderguard.js]: I prefer not to use `assert.hurts` in general. You can just check if HP equals max HP.

File: test/sim/abilities/wonderguard.js [hunk 2]
```diff
@@ -43,4 +43,12 @@ describe('Wonder Guard', function () {
 		battle.setPlayer('p2', {team: [{species: "Reshiram", ability: 'turboblaze', moves: ['fusionflare']}]});
 		assert.hurts(battle.p1.active[0], () => battle.makeChoices('move sleeptalk', 'move fusionflare'));
 	});
+
+	it('should allow first turn of Sky Drop but not the second', function () {
+		battle = common.createBattle();
+		battle.setPlayer('p1', {team: [{species: "Smeargle", ability: 'wonderguard', moves: ['sleeptalk']}]});
+		battle.setPlayer('p2', {team: [{species: "Aerodactyl", ability: 'owntempo', moves: ['skydrop']}]});
+		battle.makeChoices('move sleeptalk', 'move skydrop');
```
Comment (Zarel) [inline test/sim/abilities/wonderguard.js]: You should probably have Smeargle use a move with a side effect (like Howl), so you can make sure it didn't take effect.

Comment (Zarel) [reply test/sim/abilities/wonderguard.js]: This is, after all, the most important part of this test: You need to write a test that fails before your test, and succeeds now.

This is one of the most important parts of writing tests: Fixing confirmation bias. It's not enough for a test to pass for things you want, it also needs to fail for things you don't want.

Comment (Volco) [reply test/sim/abilities/wonderguard.js]: Okay, that's good to know for the future

File: test/sim/abilities/wonderguard.js (hunk 1 above)
Comment (Volco) [reply test/sim/abilities/wonderguard.js]: Okay, i'll do the changes tomorrow after classes, adding the side effect move (howl) and this change

**Comment by Volco** [comment]
Okay, hopefully it's good now. if changes are needed in the test file lmk. Still not great with the test files, first time i've editted them

File: test/sim/abilities/wonderguard.js [hunk 3]
```diff
@@ -43,4 +43,12 @@ describe('Wonder Guard', function () {
 		battle.setPlayer('p2', {team: [{species: "Reshiram", ability: 'turboblaze', moves: ['fusionflare']}]});
 		assert.hurts(battle.p1.active[0], () => battle.makeChoices('move sleeptalk', 'move fusionflare'));
 	});
+
+	it('should allow first turn of Sky Drop but not the second', function () {
+		battle = common.createBattle();
+		battle.setPlayer('p1', {team: [{species: "Smeargle", ability: 'wonderguard', moves: ['howl']}]});
+		battle.setPlayer('p2', {team: [{species: "Aerodactyl", ability: 'owntempo', moves: ['skydrop']}]});
+		battle.makeChoices('move howl', 'move skydrop');
+		assert.fullHP(battle.p1.active[0]);
```
Comment (Zarel) [inline test/sim/abilities/wonderguard.js]: You're not testing for the side effects of the Howl. You need to test that the Howl didn't happen (that Smeargle's Atk isn't +1).

Comment (Zarel) [reply test/sim/abilities/wonderguard.js]: You're also not testing that the second turn isn't allowed: this is only a test for the first turn.

**Comment by AnnikaCodes** [comment]
This is marked as "Done" in the kanban board, so I assume it's been fixed.

**Pull request closed without merging**